### ISSUE TITLE
Problem with error logging

### ISSUE THREAD

**Issue by Seppo-Konttinen-Solteq** [klevu/productsearch#75]
Hi.

We have version 3.2.1.

In:

vendor/klevu/module-productsearch/Model/Api/Action/Addrecords.php:prepareOtherParameters

there is error logged:

`$this->_searchHelperData->log(
    LoggerConstants::ZEND_LOG_ERR,
   __(
      '%1: Multi dimensional array provided for "other" for SKU %2 : attribute %3.',
      __METHOD__,
      $record['sku'],
      $key
   )
);`

But how it looks in log file:

`[2023-03-23T11:50:29.828253+00:00] Klevu.ERROR: %s: Multi dimensional array provided for "other" attribute for SKU $s. [] []`

which makes harder to check which product caused the error. So there is problem with sprintf functionality in this case.

Thanks,
Seppo

**Comment by klevu** [comment]
Thanks for the report.

The issue is here https://github.com/klevu/productsearch/blob/3.x/Model/Api/Action/Addrecords.php#L358
The code is using `%s`, which is fine for sprintf, but here we're using a Phrase and those placeholders should be `%1` and `%2`.

Attached  is a patch to update that. 
[klevu-productsearch-3.2.1-api-log.patch](https://github.com/klevu/productsearch/files/11050787/klevu-productsearch-3.2.1-api-log.patch)
I have tested that the patch installs correctly on 3.2.1.
Please test this on a none production env, it has **NOT** been through QA at our end.

We will get that fixed in the next release.

---

This is a composer patch and should be installed as such. 
These changes are currently going through QA and should be first installed in a none production environment and thoroughly tested to ensure there are no issues.

Patch installation instructions
```
# Create the patches directory if not already exists
mkdir patches/composer -p

# Add the patch file
vi patches/composer/klevu-productsearch-3.2.1-api-log.patch  # add the patch content provided

# Update composer.json to include patch information under "extra" section
vi composer.json

"extra": {
	"patches": {
		"klevu/module-productsearch": {
			"Product Search 3.2.1": "patches/composer/klevu-productsearch-3.2.1-api-log.patch"
		}
	}
}

# Install composer patches if not already available
# Either https://github.com/vaimo/composer-patches
composer require vaimo/composer-patches

# or https://github.com/cweagans/composer-patches
composer require cweagans/composer-patches

# Apply the patch
composer -v install

# Update the composer lock file
composer update --lock
```

**Comment by klevu** [comment]
@Seppo-Konttinen-Solteq Thank you for raising this issue. We have released the fix for this today in both `2.10.2` and `3.2.2`.
